api/api.go

```diff
@@ -196,18 +197,35 @@ type SlabsUploadRequest struct {
 	CurrentHeight uint64     `json:"currentHeight"`
 }
 
+// SlabsUploadResponse is the response type for the /slabs/upload endpoint.
+type SlabsUploadResponse struct {
+	Slab     slab.Slab         `json:"slab"`
+	Metadata []HostInteraction `json:"metadata"`
+}
+
 // SlabsDownloadRequest is the request type for the /slabs/download endpoint.
 type SlabsDownloadRequest struct {
 	Slab      slab.Slice `json:"slab"`
 	Contracts []Contract `json:"contracts"`
 }
 
+// SlabsDownloadResponse is the response type for the /slabs/download endpoint.
+type SlabsDownloadResponse struct {
+	Data     []byte            `json:"data"`
```

Comment:
JSON encoding a sector worth of `Data` adds about 1MiB of overhead, and decoding takes ~24ms on my laptop. Could `Data` be written directly to the response and the host interactions JSON encoded before or after?